debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java

```diff
@@ -826,6 +826,43 @@ public void shouldRemoveWhiteSpaceChars() throws Exception {
         assertThat(sourceTable).isEqualTo("b");
     }
 
+    @Test
+    @FixFor("DBZ-2118")
+    public void shouldCloseTxAfterTypeQuery() throws Exception {
+        String setupStmt = SETUP_TABLES_STMT;
+
+        TestHelper.execute(setupStmt);
+        Configuration.Builder configBuilder = TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.SNAPSHOT_MODE, SnapshotMode.INITIAL.getValue())
+                .with(PostgresConnectorConfig.DROP_SLOT_ON_STOP, Boolean.TRUE)
+                .with(PostgresConnectorConfig.SCHEMA_WHITELIST, "s1")
+                .with(PostgresConnectorConfig.TABLE_WHITELIST, "s1.b");
+
+        start(PostgresConnector.class, configBuilder.build());
+        assertConnectorIsRunning();
+        waitForSnapshotToBeCompleted();
+
+        TestHelper.execute("CREATE TABLE s1.b (pk SERIAL, aa isbn, PRIMARY KEY(pk));", "INSERT INTO s1.b (aa) VALUES ('978-0-393-04002-9')");
+        SourceRecords actualRecords = consumeRecordsByTopic(1);
```

Comment:
This looks like it could be race condition. Relying on the 5 ms poll in `consumeRecordsByTopic` to ensure that the connector picked up the records seems sorta flake. A specific case where this would be problematic is if a developer wanted to use a debugger on the connecter during this test. What about polling on the JMX field `totalNumberOfEventsSeen` to wait till we are sure the connector pushed the record?

Reply:
While I agree in theory, it's not a problem in practice. The poll timeout is 5 *seconds* and it's tried three times, i.e. 15 sec overall. That's more than enough even on our slow CI environment and this method doesn't show caus any troubles there (that said, we *do* have flaky tests due to similar causes, but that's not one of them). In terms of debugging, you'd typically have two breakpoints: one in the event loop and one in the main thread, right before this call. Then you can take all the time you need to step through the event loop.